### ISSUE TITLE
Easier way to observe optimizer hyperparameters

### ISSUE THREAD

**Issue by niboshi** [chainer/chainer#7230]
There is a trainer extension `chainer.training.extensions.observe_lr` for observing a learning rate, but not for general hyperparameters.
In order to observe other hyperparameters, we have to write something like this:

```py
chainer.training.extensions.observe_value(
    'weight_decay_rate',
    lambda trainer: trainer.updater.get_optimizer('main').weight_decay_rate)
```

This is not something ordinary users would like to write.
I suggest adding more easy-to-write interface, and perhaps deprecate `observe_lr`.

Possible options:

1. Update `observe_value` to allow something like `observe_value('weight_decay_rate', mode='optimizer')`
2. Add another interface like `observe_optimizer('weight_decay_rate')`
